File: backend/app/models/announcement.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, Text, Boolean, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.database import Base


class Announcement(Base):
    __tablename__ = "announcements"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    title = Column(String(255), nullable=False)
    content = Column(Text, nullable=True)
    type = Column(String(50), nullable=False, default="bulletin")
    target_campus = Column(String(50), nullable=False, default="both")
    created_by = Column(UUID(as_uuid=True), ForeignKey("users.id"))
    is_active = Column(Boolean, default=True)
    display_until = Column(DateTime, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)

    creator = relationship("User", back_populates="announcements")

File: frontend/src/types/announcement.ts
export interface Announcement {
  id: string;
  title: string;
  content: string | null;
  type: 'bulletin' | 'emergency' | 'info';
  target_campus: 'paete' | 'pagsanjan' | 'both';
  created_by: string;
  is_active: boolean;
  display_until: string | null;
  created_at: string;
}
